homeassistant/components/template/trigger_entity.py

```diff
@@ -71,3 +84,93 @@ def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
         self._process_data()
         self.async_write_ha_state()
+
+    def _render_single_template(
+        self,
+        key: str,
+        variables: dict[str, Any],
+        is_complex: bool,
```

Comment:
We don't need to pass this flag, instead, the method can check if the key is in `self._to_render_complex`